### PR TITLE
Raise exception when the ssh key is not valid

### PR DESCRIPTION
**Pull request by JosepSampe**
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: lithops/util/ssh_client.py [hunk 1]
```diff
@@ -29,13 +30,19 @@ def create_client(self, timeout=2):
...
+            self.ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            self.ssh_client.connect(self.ip_address, **self.ssh_credentials,
+                                    timeout=timeout, banner_timeout=200,
+                                    allow_agent=False)
 
-        logger.debug("{} ssh client created".format(self.ip_address))
+            logger.debug("{} ssh client created".format(self.ip_address))
+        except Exception as e:
+            pk = self.ssh_credentials.get('key_filename')
+            if pk and str(e) == 'Authentication failed.':
```
Comment (kpavel) [inline lithops/util/ssh_client.py:43]: If wrong private key specified i see exception:
`"Bad authentication type; allowed types: ['publickey']"`

**Comment by kpavel** [comment]
There some disadvantage in this approach:

If for some reason user specifies wrong `non default` private key, while having `correct` private key as `default`, lithops will work fine. 
Then, after some period, when user changes `default` key while keeping working with his `non default` private key, lithops will fail and user will be very confused why did it stop working...

**Comment by JosepSampe** [comment]
@kpavel Yes, I've already been working on this. It is fixed using `look_for_keys=False` in the ssh client. So now, only the key specified by the user in config is used.

File: lithops/util/ssh_client.py (hunk 1 above)
Comment (JosepSampe) [reply lithops/util/ssh_client.py:43]: Do you updated your code with https://github.com/lithops-cloud/lithops/pull/863/files ?

**Comment by JosepSampe** [comment]
Re-enabled IBM VPC logic to verify ssh keys. I keep logic in ssh client because it can be complementary, and because VM and EC2 backends does not have specific logic to raise exceptions in case of wrong key

**Pull request merged**